lib/Haskell/Prelude.agda

```diff
@@ -1,20 +1,76 @@
 {-# OPTIONS --no-auto-inline #-}
 module Haskell.Prelude where
 
-open import Agda.Builtin.Unit  public
-open import Agda.Builtin.List  public
-open import Agda.Builtin.Nat   public
-open import Agda.Builtin.Bool  public
-open import Agda.Builtin.Float public
-open import Agda.Builtin.Char  public
+open import Agda.Builtin.Unit         public
+open import Agda.Builtin.List         public
+open import Agda.Builtin.Nat   as Nat public hiding (_==_; _<_)
+open import Agda.Builtin.Bool         public
+open import Agda.Builtin.Float        public
+open import Agda.Builtin.Char         public
+open import Agda.Builtin.FromString   public
+import Agda.Builtin.String as Str
+open import Agda.Builtin.Strict
 
--- Type variables --
+-- Problematic features
+--  - [Partial]:  Could pass implicit/instance arguments to prove totality.
```

Comment:
This sounds reasonable, but we'll have to be careful to erase this totality information, and it will not suffice to just ignore implicit arguments for instance (since typeclasses in #3 use these). :thinking:

Reply:
Yes, so they would have to go in implicit arguments rather than instance arguments.